actors/miner/src/lib.rs

```diff
@@ -4090,14 +4090,27 @@ fn handle_proving_deadline(
 
     let state: State = rt.transaction(|state: &mut State, rt| {
         let policy = rt.policy();
-        // Vest locked funds.
-        // This happens first so that any subsequent penalties are taken
-        // from locked vesting funds before funds free this epoch.
-        let newly_vested = state
-            .unlock_vested_funds(rt.store(), rt.curr_epoch())
-            .map_err(|e| e.downcast_default(ExitCode::USR_ILLEGAL_STATE, "failed to vest funds"))?;
-
-        pledge_delta_total -= newly_vested;
+
+        // Vesting rewards for a miner are quantised to every 12 hours and we can determine what those "vesting epochs" are
+        // based on the proving period offset for the Miner.
+        // So, only do the vesting here if the current epoch is a "vesting epoch"
+        let q = QuantSpec {
+            unit: REWARD_VESTING_SPEC.quantization,
+            offset: state.current_proving_period_start(rt.policy(), curr_epoch),
```

Comment:
I think you can just use "self.proving_period_start" as the offset here.

Reply:
@anorth I was a bit confused about that. What's the fundamental difference st.current_proving_period_start() and self.proving_period_start here ? I see the proving_period_start on the state is initialised using the current_proving_period_start() call on Actor construction but what's the use of the `current_proving_period_start` methods beyond that ?

Reply:
You could use `self.proving_period_start` here even if it wasn't being kept up to date by cron. But yes the code here does seem confusing and it's possible that there's more complexity than necessary. In the past, I think proving_period_start was never updated, just treated as an offset, and so this method computed the "current" version. Also in the past, cron did not reliably run every epoch, but now it does. So we could probably simplify things here, possibly by removing that method in other places.